plugins/catalog/src/hooks/useEntity.ts

```diff
@@ -58,6 +58,8 @@ export const useEntityFromUrl = (): EntityLoadingStatus => {
  * Always going to return an entity, or throw an error if not a descendant of a EntityProvider.
  */
 export const useEntity = () => {
-  const { entity } = useContext<{ entity: Entity }>(EntityContext as any);
-  return { entity };
+  const { entity, loading, error } = useContext<EntityLoadingStatus>(
```

Comment:
@shmidt-i I noticed that since `entity` is optional in `EntityLoadingStatus` it is giving error. If we pass `{
    entity: Entity;
    loading: boolean;
    error: Error;
  }` instead of `EntityLoadingStatus` to `useContext`, the error is resolved. Should I make that change?

Reply:
Yes, let's start with that 👍

Reply:
I mean you literally just covered the `entity is not there` case :)

Reply:
Thanks. I've make the changes. Please check it